apis/core/lib/domain/csv/index.ts

```diff
@@ -0,0 +1,36 @@
+import CSVparse from 'csv-parse'
+import CSVSniffer from 'csv-sniffer'
+
+const csvDelimiters = [',', ';', '\t']
+const sniffer = new (CSVSniffer())(csvDelimiters)
+
+function parse(csv: string, options: CSVparse.Options): Promise<Array<Array<any>>> {
+  return new Promise((resolve, reject) => CSVparse(csv, options, (err, records) => {
+    if (err) {
+      reject(err)
+      return
+    }
+    resolve(records)
+  }))
+}
+
+export async function sniffParse(csv: string) {
```

Comment:
> I might move the fixtures to Minio's bucket later on.

I don't think we need that for unit tests of the `sniffParse` function. This would only be necessary if we have a request handler to re-run the sniffing. Such as in the scenario we discussed where the user would change the dialect or maybe upload a newer version of the file.

Instead I would suggest accepting a stream as input so that in the final code we will pass in an S3 stream and in tests a file stream.

```suggestion
export async function sniffParse(csv: Stream) {
```

Reply:
So `sniffParse` would take a stream, read the first chunk only, sniff and parse it, push the dialect and the chunk to the stream, then push the other chunks as they come?

Reply:
Not exactly. My idea was to save the entire file first to S3 and then open in from the bucket to do the sniffing.

Reply:
Right, that's what I was going for:

- Something to write the file to S3
- Something to read a whole file
- Something to read the beginning of a file <- this part will sniff and parse, the parsed part will be used for preview, the sniffed dialect will be stored and reused when reading the whole file